jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHandler.java

```diff
@@ -1350,27 +1411,28 @@ protected synchronized void updateMappings()
         {
             PathMappings<ServletHolder> pm = new PathMappings<>();
 
-            //create a map of paths to set of ServletMappings that define that mapping
-            HashMap<String, List<ServletMapping>> sms = new HashMap<>();
+            // Create a map of String paths (servlet url-pattern) to set of ServletMappings that define that mapping
+            HashMap<String, List<ServletMapping>> servletUrlMappings = new HashMap<>();
+
             for (ServletMapping servletMapping : _servletMappings)
             {
                 String[] pathSpecs = servletMapping.getPathSpecs();
                 if (pathSpecs != null)
                 {
                     for (String pathSpec : pathSpecs)
                     {
-                        List<ServletMapping> mappings = sms.computeIfAbsent(pathSpec, k -> new ArrayList<>());
+                        List<ServletMapping> mappings = servletUrlMappings.computeIfAbsent(pathSpec, k -> new ArrayList<>());
                         mappings.add(servletMapping);
                     }
                 }
             }
 
-            //evaluate path to servlet map based on servlet mappings
-            for (String pathSpec : sms.keySet())
+            // evaluate path to servlet map based on servlet mappings
+            for (String pathSpec : servletUrlMappings.keySet())
```

Comment:
PathMappings is add order dependent, so I don't really like the fact that we add all the `servletUrlMappings` and them below add all the bespoke path mappings.      Is it possible to have a single loop through `_servletMappings` that adds both standard mappings and bespoke mappings in exactly the order of the mappings themselves?